File: codegen/results/v001/qwen2.5-coder_7b/20260409_102956/python/003_fibonacci/solution.py
import sys

def fibonacci_up_to_n(n):
    if n < 1:
        return []
    
    fib_sequence = [1, 1]
    while True:
        next_fib = fib_sequence[-1] + fib_sequence[-2]
        if next_fib > n:
            break
        fib_sequence.append(next_fib)
    
    return fib_sequence

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python script.py <integer>")
        sys.exit(1)
    
    try:
        N = int(sys.argv[1])
    except ValueError:
        print("Please provide a valid integer.")
        sys.exit(1)
    
    fib_numbers = fibonacci_up_to_n(N)
    for number in fib_numbers:
        print(number)

File: codegen/results/v001/qwen2.5-coder_7b/20260409_102956/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Println("Usage: go run fib.go <integer>")
		return
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil || n < 1 {
		fmt.Println("Please provide a positive integer.")
		return
	}

	var a, b int = 0, 1
	for b <= n {
		a, b = b, a+b
		fmt.Println(a)
	}
}